syntax = "proto3";

package yandex.cloud.priv.marketplace.licenseserver.v1.license.console;

import "yandex/cloud/api/operation.proto";
import "yandex/cloud/priv/operation/operation.proto";
import "yandex/cloud/priv/validation.proto";
import "yandex/cloud/priv/marketplace/licenseserver/v1/license/instance.proto";

option go_package = "a.yandex-team.ru/cloud/bitbucket/private-api/yandex/cloud/priv/marketplace/licenseserver/v1/license/console;console";
option java_package = "CIS";

service InstanceService {
    rpc Get(GetInstanceRequest) returns (Instance);

    rpc List(ListInstanceRequest) returns (ListInstanceResponse);

    rpc Create(CreateInstanceRequest) returns (operation.Operation) {
        option (yandex.cloud.api.operation) = {
            metadata: "CreateInstanceMetadata"
            response: "Instance"
        };
    }

    rpc Cancel(CancelInstanceRequest) returns (operation.Operation) {
        option (yandex.cloud.api.operation) = {
            metadata: "CancelInstanceMetadata"
            response: "Instance"
        };
    }
}

message GetInstanceRequest {
    string id = 1 [(required) = true];
}

message ListInstanceRequest {
    string cloud_id = 1 [(required) = true];
    int64 page_size = 2;
    string page_token = 3;
}

message ListInstanceResponse {
    repeated Instance instances = 1;
    string next_page_token = 2;
}

message CreateInstanceRequest {
    string template_id = 1 [(required) = true];
    string cloud_id = 2 [(required) = true];
    string name = 3;
}

message CreateInstanceMetadata {
    string instance_id = 1;
}

message CancelInstanceRequest {
    string id = 1 [(required) = true];
}

message CancelInstanceMetadata {
    string instance_id = 1;
}
